services/src/test/scala/cromwell/pipeline/service/GitLabProjectVersioningTest.scala

```diff
@@ -183,7 +183,69 @@ class GitLabProjectVersioningTest extends AsyncWordSpec with ScalaFutures with M
           .getFile(project, path, Some(version))
           .map(_ shouldBe Left(VersioningException("Exception. Response status: 404")))
       }
+    }
 
+    "getFileCommits" should {
+      val dummyCommitJson: String = s"[${Json.stringify(Json.toJson(dummyFileCommit))}]"
+      val path: Path = Paths.get("tmp/foo.txt")
+      def request(project: Project) =
+        mockHttpClient.get(
+          url = gitLabConfig.url + "projects/" + project.repository.get.value + "/repository/files/" + path.toString,
```

Comment:
The same about path encoding and btw use string interpolation for concatination a lot of strings

Reply:
Yeah, concatination slightly slow down your performance.

Reply:
Changed